src/inmanta/protocol/openapi/converter.py

```diff
@@ -150,62 +149,109 @@ def get_swagger_html(self, openapi_spec: str) -> str:
+        if len(schema["anyOf"]) == 1:
+            anyOfList = schema["anyOf"]
+            del schema["anyOf"]
+            schema.update(anyOfList[0])
+
+    # TODO: clean up! Probably better to do this with a validator on Schema
+    if schema.get("items", None) is not None:
+        schema["items"] = nullable_item(schema["items"])
+    if schema.get("additionalProperties", None) is not None:
+        if not isinstance(schema["additionalProperties"], bool):
+            schema["additionalProperties"] = nullable_item(schema["additionalProperties"])
+    for sequence_subfield in ("anyOf", "allOf", "oneOf"):
+        if schema.get(sequence_subfield, None) is not None:
+            schema[sequence_subfield] = [nullable_item(sub) for sub in schema[sequence_subfield]]
+    if schema.get("properties", None) is not None:
+        for prop, sub in schema["properties"].items():
+            schema["properties"][prop] = nullable_item(sub)
+
+    return schema
+
+
+def generate_nullable(schema: dict[str, Any]) -> dict[str, Any]:
+    """Generate nullable based on the union types. In case there is a union (anyOf) with null, nullable will be set and the
+    anyOf list will be modified.
+    """
+    for prop, sub in schema.get("properties", {}).items():
+        schema["properties"][prop] = nullable_item(sub)
+
+    return schema
+
+
 class OpenApiTypeConverter:
     """
     Lookup for OpenAPI types corresponding to python types
     """
 
     def __init__(self) -> None:
         self.components = Components(schemas={})
-        self.ref_prefix = "#/components/schemas/"
-        self.ref_regex = re.compile(self.ref_prefix + r"(.*)")
-
-        # Applying some monkey patching to pydantic
-        patch_pydantic_field_type_schema()
+        self._pydantic_ref_key = "$defs"
+        self.openapi_ref_prefix = "#/components/schemas/"
+        self._ref_regex = re.compile(re.escape(self.openapi_ref_prefix) + r"(.*)")
 
     def get_openapi_type_of_parameter(self, parameter_type: inspect.Parameter) -> Schema:
         schema = self.get_openapi_type(parameter_type.annotation)
         if parameter_type.default is not inspect.Parameter.empty:
             schema.default = parameter_type.default
         return schema
 
-    def _handle_pydantic_model(self, type_annotation: Type, by_alias: bool = True) -> Schema:
-        # JsonSchema stores the model (and sub-model) definitions at #/definitions,
-        # but OpenAPI requires them to be placed at "#/components/schemas/"
-        # The ref_prefix changes the references, but the actual schemas are still at #/definitions
-        schema = model_schema(type_annotation, by_alias=by_alias, ref_prefix=self.ref_prefix)
-        if "definitions" in schema.keys():
-            definitions: Dict[str, Dict[str, object]] = schema.pop("definitions")
+    def _handle_pydantic_model(self, type_annotation: BaseModel, by_alias: bool = True) -> Schema:
```

Comment:
It seems the `by_alias` parameter is not used, is this intentional ?

Reply:
No, good catch, will fix